### ISSUE TITLE
Uncaught TypeError: ace.define is not a function

### ISSUE THREAD

**Issue by PsyTae** [ajaxorg/ace-builds#78]
I am not using RequireJS, but using the [src](https://github.com/ajaxorg/ace-builds/tree/e94cb3c7ffccfb2e565cc857f07a988c0989e527/src) to load ace and we are loading it in this order in our html.

```
    <script src="../scripts/aceSrc/ace.js" type="text/javascript" charset="utf-8"></script>
    <script src="../scripts/aceSrc/ext-language_tools.js" type="text/javascript" charset="utf-8"></script>
    <script src="../scripts/aceSrc/ext-modelist.js" type="text/javascript" charset="utf-8"></script>

    <script src="../scripts/webIntellisense/webintellisense.js"></script>
    <script src="../scripts/webIntellisense/webintellisense-ace.js"></script>
    <link href="../styles/webIntellisense/webintellisense.css" rel="stylesheet" />
```
However right away we get an error
`Uncaught TypeError: ace.define is not a function(anonymous function) @ webintellisense-ace.js:1`

I checked and in the ace.js file line 49 `var define =` is present so I am not sure why we are getting this error.

We have a working ACE Editor in the page, and are trying to add the WebIntellisense module but have not added in any of the JS yet into the page to implement webIntellisense other than the includes. The very first line of the webintellisense-ace.js script is:
```
ace.define('ace/intellisense',
    ['require', 'exports', 'module', 'ace/keyboard/hash_handler'],
    function (require, exports, module)
    {
```
This appears to meet the qualifications of the function in ace.js for `var define =`

Are we loading the scripts out of order?

**Comment by nightwing** [comment]
Looks like ace.define is missing in https://github.com/ajaxorg/ace-builds/tree/master/src

Use ace.js from https://github.com/ajaxorg/ace-builds/tree/master/src-noconflict, 
I'll fix the first version in the next release.

**Comment by PsyTae** [comment]
using the ace.js from https://github.com/ajaxorg/ace-builds/tree/master/src-noconflict, caused  MANY more errors.

Uncaught ReferenceError: define is not defined(anonymous function) @ ext-language_tools.js:1
ext-modelist.js:1 Uncaught ReferenceError: define is not defined(anonymous function) @ ext-modelist.js:1

Uncaught ReferenceError: define is not defined(anonymous function) @ ext-modelist.js:1
ace.js:3543 misspelled option "enableSnippets"

misspelled option "enableSnippets"
misspelled option "enableBasicAutocompletion"
misspelled option "enableLiveAutocompletion"

Uncaught TypeError: Cannot read property 'getModeForPath' of undefined(anonymous function) @ editPageCSS.cfm?fileIdIn=65D33811BF75969C3010BCDE089A7C51:910
theme-monokai.js:1 Uncaught ReferenceError: define is not defined(anonymous function) @ theme-monokai.js:1

Uncaught ReferenceError: define is not defined(anonymous function) @ theme-monokai.js:1
ace.js:16190 Uncaught TypeError: Cannot read property 'cssClass' of undefinedafterLoad @ ace.js:16190(anonymous function) @ ace.js:3711(anonymous function) @ ace.js:3710_require @ ace.js:88req @ ace.js:138afterLoad @ ace.js:3706s.onload.s.onreadystatechange @ ace.js:3352

Uncaught TypeError: Cannot read property 'cssClass' of undefinedafterLoad @ ace.js:16190(anonymous function) @ ace.js:3711(anonymous function) @ ace.js:3710_require @ ace.js:88req @ ace.js:138afterLoad @ ace.js:3706s.onload.s.onreadystatechange @ ace.js:3352

**Comment by PsyTae** [comment]
should I switch all of my files in my aceSrc folder to the no-conflict versions?

**Comment by nightwing** [comment]
yes

**Comment by PsyTae** [comment]
switching all files to the src-noconflict branch appears to have fixed the issue.  We are no longer getting the ace.define is not a function error.

**Comment by blurymind** [comment]
I still get this issue even with the fix suggested here

**Comment by ttaylor29** [comment]
nightwing:

I'm using these 4 with dev express controls and I get the ace is not defined error:

                    "~/Scripts/devexpress/ace.js",
                    "~/Scripts/devexpress/ext-language_tools.js",
                    "~/Scripts/devexpress/theme-ambiance.js",
                    "~/Scripts/devexpress/theme-dreamweaver.js"));

I pulled all the files for the above 4 from:

https://github.com/ajaxorg/ace-builds/tree/master/src-noconflict,

And I still get the error. 

Any ideas why I still get this?

ext-language_tools.js:1 Uncaught ReferenceError: ace is not defined
    at ext-language_tools.js:1

**Comment by nightwing** [comment]
Could you tell more about how to reproduce the error, https://github.com/ajaxorg/ace-builds/blob/master/src-noconflict/ace.js#L152 is supposed to define global ace.

**Comment by ttaylor29** [comment]
Hey nightwing:

I realized that ent_language_tools was being added FIRST before ace.js. 

I had to bundle it up in its own bundle and then it moved it UNDER ace.js.

Even if I placed: 

`       "~/Scripts/devexpress/ace.js",
                "~/Scripts/devexpress/ext-language_tools.js",
                "~/Scripts/devexpress/theme-ambiance.js",
                "~/Scripts/devexpress/theme-dreamweaver.js"));`

Under the SAME bundle, it still would put ext-language_tools.js" before ace.js.

Everything is good now. 

(Just other errors not tied to this!)
